extensions/ql-vscode/src/databases-ui.ts

```diff
@@ -361,13 +362,34 @@ export class DatabaseUI extends DisposableObject {
     }
 
     if (byFolder) {
+      const fixedUri = await this.fixDbUri(uri);
       // we are selecting a database folder
-      return await this.setCurrentDatabase(uri);
+      return await this.setCurrentDatabase(fixedUri);
     }
     else {
       // we are selecting a database archive. Must unzip into a workspace-controlled area
       // before importing.
       return await importArchiveDatabase(uri.toString(true), this.databaseManager, this.storagePath);
     }
   }
+
+  /**
+   * Perform some heuristics to ensure a proper database location is chosen.
+   *
+   * 1. If the selected URI to add is a file, choose the containing directory
+   * 2. If the selected URI is a directory matching db-*, choose the containing directory
+   * 3. choose the current directory
+   *
+   * @param uri
+   */
+  private async fixDbUri(uri: Uri): Promise<Uri> {
+    let dbPath = uri.fsPath;
+    if ((await fs.stat(dbPath)).isFile()) {
```

Comment:
Does this break selecting a `.zip` archive of a db? If not, maybe refine the comment slightly to emphasize that we're only talking about the codepath that is concerned with intentionally selecting a database *directory*.

Reply:
Yeah, oops, that definitely is the case here. Just requesting slightly more context in the doc comment then.